File: ScheduledTask/Scheduled.cs
namespace ScheduledTask
{
    public class Scheduled
    {
        public delegate void TaskUpdated(Task task, int index);
        public TaskUpdated? UpdateTaskCallback;

        private List<Task> taskList;

        public Scheduled()
        {
            taskList = new List<Task>();
        }

        public void AddTask(Task task)
        {
            taskList.Add(task);
        }

        public List<Task> GetDueTasks(DateTime now)
        {
            List<Task> dueTasks = new List<Task>();
            foreach (Task task in taskList)
            {
                if (task.ExecutionTime.Hour == now.Hour && task.ExecutionTime.Minute == now.Minute)
                {
                    dueTasks.Add(task);
                }
            }
            foreach (Task task in dueTasks)
            {
                taskList.Remove(task);
            }
            return dueTasks;
        }

        public bool HasTasks()
        {
            return taskList.Count > 0;
        }

        public void RemoveTask(Task task)
        {
            taskList.Remove(task);
        }

        public void UpdateReminder()
        {
            DateTime now = DateTime.Now;
            List<Task> dueTasks = GetDueTasks(now);
            foreach (Task task in dueTasks)
            {
                MessageBox.Show(task.Description, "Нагадування");
            }
        }
    }
}

File: Java/SheduledTaskJava/src/Scheduled.java
import javax.swing.*;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

class Scheduled {
    private final List<Task> taskList;

    public Scheduled() {
        taskList = new ArrayList<>();
    }

    public void addTask(Task task) {
        taskList.add(task);
    }

    public List<Task> getDueTasks() {
        List<Task> dueTasks = new ArrayList<>();
        LocalDateTime now = LocalDateTime.now();
        for (Task task : taskList) {
            if (task.getExecutionTime().isBefore(now)) {
                dueTasks.add(task);
            }
        }
        taskList.removeAll(dueTasks);
        return dueTasks;
    }

    public boolean hasTasks() {
        return !taskList.isEmpty();
    }

    public void removeTask(Task task) {
        taskList.remove(task);
    }

    public void updateReminder() {
        List<Task> dueTasks = getDueTasks();
        for (Task task : dueTasks) {
            JOptionPane.showMessageDialog(null, task.getDescription(), "Reminder", JOptionPane.INFORMATION_MESSAGE);
        }
    }

    public List<Task> getTaskList() {
        return taskList;
    }

    public void setUpdateTaskCallback(TaskUpdated callback) {
    }

    public interface TaskUpdated {
        void updateTask(Task task, int index);
    }
}